jcl/src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -2759,7 +2759,7 @@ public synchronized int offsetByCodePoints(int start, int codePointCount) {
 		if (String.COMPACT_STRINGS && count >= 0) {
 			int index = start + codePointCount;
 
-			if (index >= currentLength) {
+			if ((index >= currentLength) || (index < 0)) {
```

Comment:
This is not consistent with `String` - I think `>=` should be `>`; likewise in `StringBuilder`.
It would be good to have tests for the boundary conditions.